### PR TITLE
Initial code

### PR DESCRIPTION
**Pull request by mangelats**
Extends #1

### CODE REVIEW

File: data_access_log/admin.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import django
+from django.contrib import admin
+
+from .models import AccessLog
```
Comment (manelclos) [inline data_access_log/admin.py]: DataAccessLog

File: data_access_log/admin.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+
+@admin.register(AccessLog)
+class AccessLogAdmin(admin.ModelAdmin):
```
Comment (manelclos) [inline data_access_log/admin.py]: DataAccessLogAdmin

**Review by manelclos — COMMENTED** [review]
please review comments

File: data_access_log/decorators.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    dispatch = viewset.dispatch
+    viewset_name = type_name(viewset)
+    model_name = type_name(viewset_model(viewset))
+
+    @wraps(dispatch)
+    def wrapper(other, request, *args, **kwargs):
+        from .models import DataAccessLog
+
+        response = dispatch(other, request, *args, **kwargs)
+        print(request.user)
```
Comment (abusquets) [inline data_access_log/decorators.py]: You should remove this print.

**Review by abusquets — CHANGES_REQUESTED** [review]
Remove print

File: data_access_log/admin.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+@admin.register(DataAccessLog)
+class DataAccessLogAdmin(admin.ModelAdmin):
+    list_display = ('model', 'viewset', 'user', 'timestamp')
+    list_display_links = ('model', 'viewset', 'timestamp')
+
+    fields = ('model', 'viewset', 'timestamp', 'user', 'url', 'response')
+    readonly_fields = fields
+
+    def has_view_permission(self, request, obj=None):
+        return True
```
Comment (manelclos) [inline data_access_log/admin.py]: please do as in has_change_permission below

File: data_access_log/models.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+
+class DataAccessLog(models.Model):
+    """Model that contains all the logged information."""
+    model = models.CharField('Accessed Model', max_length=100, null=False, blank=False)
+    viewset = models.CharField('Accessed ViewSet', max_length=100, null=False, blank=False)
+    timestamp = models.DateTimeField('Acess timestamp', auto_now_add=True, editable=False, null=False, blank=False)
```
Comment (manelclos) [inline data_access_log/models.py]: Acess -> Access

File: data_access_log/models.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    user = models.CharField('Accessing user', max_length=255, null=False, blank=False)
```
Comment (manelclos) [inline data_access_log/models.py]: Accessing user -> User

File: data_access_log/models.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    url = models.CharField('URL', max_length=255, null=False, blank=False)
+    response = models.TextField('Response', null=False, blank=False)
```
Comment (manelclos) [inline data_access_log/models.py]: Response -> Response data or Response content

File: data_access_log/models.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+    def save(self, *args, **kwargs):
+        # Only creation allowed
```
Comment (manelclos) [inline data_access_log/models.py:15]: stops saving silently? should produce some error

**Review by manelclos — CHANGES_REQUESTED** [review]
Please review comments

File: data_access_log/admin.py (hunk 4 above)
Comment (mangelats) [reply data_access_log/admin.py]: Not necessary. Versions before 2.1 will ignore it. I could add a comment to be more clear about it.

File: data_access_log/models.py (hunk 8 above)
Comment (mangelats) [reply data_access_log/models.py:15]: It is done in the official documentation: https://docs.djangoproject.com/en/2.1/topics/db/models/#overriding-predefined-model-methods

File: data_access_log/admin.py (hunk 4 above)
Comment (manelclos) [reply data_access_log/admin.py]: I mean, it is visible by default on 2.1. It should only be visible to users with the right permission added. Maybe just don't include it, as has_change_permission is already handling it properly

File: data_access_log/models.py (hunk 8 above)
Comment (manelclos) [reply data_access_log/models.py:15]: Yep, strange. Ok!

**Review by manelclos — APPROVED** [review]

**Review by abusquets — APPROVED** [review]

**Pull request merged**